### ISSUE TITLE
Getting java.lang.AbstractMethodError when using spark 2.4.0. 

### ISSUE THREAD

**Issue by chen7572** [linkedin/scanns#8]
The package was compiled okay and running file when the cluster I was using had spark version of 2.3.0. But recently the cluster I am using was upgraded and uses spark 2.4.0. Then the package no longer works and keeps giving an error as below: 

```java.lang.AbstractMethodError: com.linkedin.nn.algorithm.JaccardMinHashNNS.org$apache$spark$ml$param$Params$_setter_$paramMap_$eq(Lorg/apache/spark/ml/param/ParamMap;)V```

I did some searching online and think this might be an incompatibility issue with the version of spark, so I changed the spark version in ```build.gradle``` to 2.4.0 and only kept `targetScalaVersion '2.11.8'` in the ```settings.gradle``` file. The build was successful, but when using it on my cluster, I still get the same error as above. 

I wonder if anyone has insight on this? Thank you for your help!

**Comment by rjof** [comment]
+1

**Comment by chen7572** [comment]
@rjof A fork by Pratheekrebala was updated more recently to use `scanns` with spark 2.4.0. In addition to specifying the spark and targetScalaVersion, we also need to change the dependencies. Here is a [link](https://github.com/pratheekrebala/scanns/commit/b61c79c683d0c6964131ce284edcdd181802f831) to the changes that pratheekrebala made:
I was able to get mine to work after modifying the dependencies. Hope this works for you too!
